static-analysis-plugin/plugin/src/test/groovy/com/novoda/test/TestProject.groovy

```diff
@@ -0,0 +1,111 @@
+package com.novoda.test
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+
+abstract class TestProject {
+    private static final Closure<String> EXTENSION_TEMPLATE = { TestProject project ->
+        """
+staticAnalysis {
+    ${formatPenalty(project)}
+}
+"""
+    }
+
+    private final File projectDir
+    private final GradleRunner gradleRunner
+    private final Closure<String> template
+    Map<String, List<File>> sourceSets = [main: []]
+    String penalty
+
+    TestProject(Closure<String> template) {
+        this.template = template
+        this.projectDir = createProjectDir("${System.currentTimeMillis()}")
+        this.gradleRunner = GradleRunner.create()
+                .withProjectDir(projectDir)
+                .withDebug(true)
+                .withPluginClasspath()
```

Comment:
This is kind of magic. Applying the `java-gradle-plugin` to the `plugin` project we can ensure its output is easily added to the integration tests runtime via `GradleRunner.withPluginClasspath()`. No more hacks are needed.